lib/modules/manager/pep621/utils.ts

```diff
@@ -0,0 +1,101 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import { regEx } from '../../../util/regex';
+import { PypiDatasource } from '../../datasource/pypi';
+import type { PackageDependency } from '../types';
+import type { Pep508ParseResult } from './types';
+
+const pep508Regex = regEx(
+  /^(?<packageName>[A-Z0-9._-]+)\s*(\[(?<extras>[A-Z0-9,._-]+)\])?\s*(?<currentValue>[^;]+)?(;\s*(?<marker>.*))?/i
+);
+
+export function parsePEP508(
+  value: string | null | undefined
+): Pep508ParseResult | null {
+  if (is.nullOrUndefined(value)) {
+    return null;
+  }
+
+  const regExpExec = pep508Regex.exec(value);
+  if (
+    is.nullOrUndefined(regExpExec) ||
+    is.nullOrUndefined(regExpExec?.groups)
+  ) {
+    logger.trace(`Pep508 could not be extracted`);
+    return null;
+  }
+
+  const result: Pep508ParseResult = {
+    packageName: regExpExec.groups.packageName,
+  };
+  if (is.nonEmptyString(regExpExec.groups.currentValue)) {
+    result.currentValue = regExpExec.groups.currentValue;
+  }
+  if (is.nonEmptyString(regExpExec.groups.marker)) {
+    result.marker = regExpExec.groups.marker;
+  }
+  if (is.nonEmptyString(regExpExec.groups.extras)) {
+    result.extras = regExpExec.groups.extras.split(',');
+  }
+
+  return result;
+}
+
+export function pep508ToPackageDependency(
+  depType: string,
+  value: string
+): PackageDependency | null {
+  const parsed = parsePEP508(value);
+  if (is.nullOrUndefined(parsed)) {
+    return null;
+  }
+
+  const dep: PackageDependency = {
+    packageName: parsed.packageName,
+    depName: parsed.packageName,
+    datasource: PypiDatasource.id,
+    depType,
+  };
+
+  if (is.nullOrUndefined(parsed.currentValue)) {
+    dep.skipReason = 'any-version';
```

Comment:
```suggestion
    dep.skipReason = 'no-version';
```
i think this is the right reason? we use that on other managers. 🤔

Reply:
My understanding is that: 

`any-version` is used if there is an unlimited range e.g. `*` or no range at all like here.    
`no-version` if there should be field with the version, but it is missing. 

But I'm not aware that we have document that anywhere...

Reply:
Definitely * maps to any-version. I don't recall if there's a really a need for `no-version` too but it seems in wide use. It could also be renamed to "unspecified-version"